pallets/pool-system/src/mock.rs

```diff
@@ -385,7 +384,40 @@ parameter_types! {
 	pub const PoolDeposit: Balance = 1 * CURRENCY;
 }
 
+pub struct LiquidityAndPoolAdmin;
+impl EnsureOriginWithArg<RuntimeOrigin, PoolId> for LiquidityAndPoolAdmin {
+	type Success = ();
+
+	#[cfg(feature = "runtime-benchmarks")]
+	fn try_origin(_: RuntimeOrigin, _: &PoolId) -> Result<Self::Success, RuntimeOrigin> {
+		Ok(())
+	}
+
+	#[cfg(not(feature = "runtime-benchmarks"))]
+	fn try_origin(o: RuntimeOrigin, _: &PoolId) -> Result<Self::Success, RuntimeOrigin> {
+		<RuntimeOrigin as Into<Result<RawOrigin<AccountId>, RuntimeOrigin>>>::into(o).and_then(
+			|r| match r {
+				RawOrigin::Root => Ok(()),
+				RawOrigin::Signed(account) => {
+					if account == DEFAULT_POOL_OWNER {
+						Ok(())
+					} else {
+						Err(RawOrigin::Signed(account).into())
+					}
+				}
+				RawOrigin::None => Err(RawOrigin::None.into()),
+			},
+		)
+	}
```

Comment:
Why do we need this special implementation? Do not we know always `Ok()` for testing?

Reply:
Mmmh. Good point. If this is covered in the integration test, probably not.